src/components/layers/drawing/drawInteraction.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { useMap } from '../../../contexts/mapContext/mapContext'
+import Draw from 'ol/interaction/Draw'
+import VectorSource from 'ol/source/Vector'
+import VectorLayer from 'ol/layer/Vector'
+import { Box, Button, Typography } from '@mui/material'
+import { getArrayOfAllLayers, getArrayOfVectorLayersWithoutDrawing } from '../getArrayOfLayers'
+import featuresCounter from './featuresCounter'
+
+const DrawInteraction = () => {
+    const mapRef = useRef(useMap())

```

Comment:
Once again, contemplate utilizing `const map` or `const { map }` directly from `useMap()` without involving the reference